azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/CSAgentPoolProfile.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.compute;
+
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.ChildResource;
+import com.microsoft.azure.management.resources.fluentcore.model.Attachable;
+import com.microsoft.azure.management.resources.fluentcore.model.HasInner;
+
+/**
+ * An client-side representation for a container service agent pool.
+ */
+@Fluent
+public interface CSAgentPoolProfile extends
+    ChildResource<ContainerService>,
+    HasInner<ContainerServiceAgentPoolProfile> {
+
+    /**
+     * @return the number of agents (VMs) to host docker containers.
+     * Allowed values must be in the range of 1 to 100 (inclusive).
+     * The default value is 1.
+     */
+    int count();
+
+    /**
+     * @return size of agent VMs.
+     */
+    ContainerServiceVMSizeTypes vmSize();
+
+    /**
+     * @return DNS prefix to be used to create the FQDN for the agent pool.
+     */
+    String dnsLabel();
+
+    /**
+     * @return FDQN for the agent pool.
+     */
+    String fqdn();
+
+
+    // Fluent interfaces
+
+    /**
+     * The entirety of a container service agent pool definition as a part of a parent definition.
+     *
+     * @param <ParentT> the return type of the final {@link Attachable#attach()}

```

Comment:
since the at-link reference does not convert well into other languages, I've been replacing occurrences of these wherever I could find them with this instead:
`@param <ParentT> the stage of the application gateway definition to return to after attaching this definition`
replace "application gateway" with the parent name.